pages/api/docs/userSelectedRepos.ts

```diff
@@ -0,0 +1,78 @@
+	console.info("[UserSelectedRepos] Saving setup info in db...");
+	const jsonBody = req.body;
+    const session = await getServerSession(req, res, authOptions);
+	if (!session?.user.id) {
+		const eventProperties = { response_status: 401 };
+        rudderStackEvents.track("absent", "", 'user-selected-repos', { type: 'HTTP-401', eventStatusFlag: 0, eventProperties });
+		return res.status(401).json({ error: 'Unauthenticated' });
+	}
+    const userId = session.user.id;
+    const provider = jsonBody.info.length > 0 ? jsonBody.info[0].provider : null;
+    if (!Array.isArray(jsonBody.info) || !jsonBody.installationId || !provider) {
+		console.error("[UserSelectedRepos] Invalid request body", jsonBody);
+		res.status(400).json({ error: "Invalid request body" });
+		const eventProperties = { response_status: 400 };
+		rudderStackEvents.track(userId, "", 'user-selected-repos', { type: 'HTTP-400', eventStatusFlag: 0, eventProperties });
+		return;
+	}
+	const event_properties = {
+		repo_provider: provider,
+		topic_name: jsonBody.installationId || "",
+		is_pat: jsonBody.isPublish || false,
+	};
+
+	try {
+		await removePreviousSelections(jsonBody.installationId, provider);
+	} catch (err) {
+		console.error(`[UserSelectedRepos] Unable to remove previous installations for ${jsonBody.installationId}`, err);
+		res.status(500).json({ "error": "Internal Server Error" });
+		const eventProperties = { ...event_properties, response_status: 500 };
+		rudderStackEvents.track(userId, "", 'user-selected-repos', { type: 'HTTP-500', eventStatusFlag: 0, eventProperties });
+		return;
+	}
+	const allSelectedReposPromises = [];
+	for (const ownerInfo of jsonBody.info) {
+		const setupReposArgs: SetupReposArgs = {
+			repo_owner: ownerInfo.owner,
+			repo_provider: provider,
+			repo_names: ownerInfo.repos,
+			install_id: jsonBody.installationId
+		}
+		const saveSelectedReposPromises = saveSelectedReposInDb(setupReposArgs, userId)
+			.catch((err) => {
+				console.error("[UserSelectedRepos] Unable to save setup info, ", err);
+				const eventProperties = { ...event_properties, response_status: 500, repo_owner: ownerInfo.owner, repos: ownerInfo.repos };
+				rudderStackEvents.track(userId, "", 'user-selected-repos', { type: 'HTTP-500', eventStatusFlag: 0, eventProperties });	
+			});
+		allSelectedReposPromises.push(saveSelectedReposPromises);
+	}
+	await Promise.all(allSelectedReposPromises).then(async () => {
+		console.info("[UserSelectedRepos] All selected repos saved succesfully...")
+		if (jsonBody.isPublish) {
+			const res = await publishMessage(jsonBody.installationId, jsonBody.info, "PATSetup");
+			console.info(`[UserSelectedRepos] Published msg to ${jsonBody.installationId}`, res);
+			const eventProperties = { ...event_properties, response_status: 500 };
+			rudderStackEvents.track(userId, "", 'user-selected-repos', { type: 'HTTP-500', eventStatusFlag: 1, eventProperties });
```

Comment:
This needs to be discussed. In such cases where the API doesn't return a response to user, then what should be the value of `response_status` event property? Shall we even include that? the same has been done in `/dpu/setup` file as well